.travis.yml

```diff
@@ -40,16 +40,17 @@ before_script:
   - ccache --zero-stats --max-size=3G
 
 env:
-  - CI_TARGETS="all ext"        # Job1: Build everything.
-  - CI_TARGETS="test"           # Job2: Test libraries. #### ext_test? adds 5min compile, runs 1 test.
+  - CI_TARGETS="all"        # Job1: Build everything.
+  - CI_TARGETS="ext"        # Job2: Build everything.
```

Comment:
I'd be surprised if that were the issue, because surely 'make -j' would know that two identical tasks generated from different top-level targets were identical.  I thought it was smart.  But I guess we'll see.

Reply:
It makes sense.

```
all:
    $(MAKE) checkversion
    $(MAKE) kaldi.mk
    $(MAKE) mklibdir
    $(MAKE) subdirs
    $(MAKE) -C matrix test
    -echo Done
```

```
ext: ext_depend $(SUBDIRS) $(EXT_SUBDIRS)
    -echo Done
```

`ext` depends explicitly on SUBDIRS in its Makefile targets, while `all` launches a new make subprocess to `$(MAKE) subdirs` to fulfill that target.

The subprocess has no idea about the `ext` target being run by the parent process, so that's why we have cases where optimization.o (for example)is created twice.

I'm not sure why `all` is defined the way that it is, but I am fairly certain that this change will fix our Travis CI problems.

Reply:
It's because of PR #1409. We need to run the checkversion and kaldi.mk targets before everything else for correctness. The ordering wouldn't be captured if we just did:

```
all: checkversion kaldi.mk mklibdir subdirs
```

Because -j can begin running those targets in any order.